### ISSUE TITLE
build fails with ghc-9.6

### ISSUE THREAD

**Issue by juhp** [parsonsmatt/persistent-discover#10]
```
       Building library for persistent-discover-0.1.0.6..                                                                [51/223668]
       [1 of 3] Compiling Database.Persist.Discover.Exe                                                                             
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/persistent-discover-0.1.0.6-399758a252e515c57db0e9a5b8308733c55fb1ed64a390e20c7a37b7e387cc69/src/Database/Persist/Discover/Exe.hs:114:15: error: [GHC-88464]                                                              
           Variable not in scope:                                                                                                   
             filterM :: (FilePath -> IO Bool) -> [FilePath] -> IO (t0 FilePath)                                                     
           Suggested fix: Perhaps use ‘filter’ (imported from Data.List)                                                            
           |                                                                                                                        
       114 |       dirs <- filterM (doesDirectoryExist . (baseDir </>)) c >>= mapM go                                               
           |               ^^^^^^^                                                                                                  
                                                                                                                                    
       /var/stackage/work/unpack-dir/unpacked/persistent-discover-0.1.0.6-399758a252e515c57db0e9a5b8308733c55fb1ed64a390e20c7a37b7e3
87cc69/src/Database/Persist/Discover/Exe.hs:115:16: error: [GHC-88464]                                                              
           Variable not in scope:                                                                                                                filterM :: (FilePath -> IO Bool) -> [FilePath] -> IO [FilePath]                                                        
           Suggested fix: Perhaps use ‘filter’ (imported from Data.List)                                                            
           |                                                                                                                        
       115 |       files <- filterM (doesFileExist . (baseDir </>)) c                                                               
           |                ^^^^^^^                                                                                                 
                                                                  
       /var/stackage/work/unpack-dir/unpacked/persistent-discover-0.1.0.6-399758a252e515c57db0e9a5b8308733c55fb1ed64a390e20c7a37b7e3
87cc69/src/Database/Persist/Discover/Exe.hs:178:5: error: [GHC-88464]
           Variable not in scope: guard :: Bool -> [a0]           
           |                                                      
       178 |     guard (extension == ".hs" || extension == ".lhs")                                                                  
           |     ^^^^^                                            
                                                                  
       /var/stackage/work/unpack-dir/unpacked/persistent-discover-0.1.0.6-399758a252e515c57db0e9a5b8308733c55fb1ed64a390e20c7a37b7e3
87cc69/src/Database/Persist/Discover/Exe.hs:202:13: error: [GHC-88464]
           Variable not in scope: guard :: Bool -> Maybe a1                                                                         
           |                                                      
       202 |             guard $ all isValidModuleName (x : xs)                                                                     
           |             ^^^^^
```

**Comment by alaendle** [comment]
Seems to be solved with 0.1.0.7.

**Comment by parsonsmatt** [comment]
Yup! Thanks for reporting the issue and for pinging me on it.
